plugins/elasticsearch/src/dsl.ts

```
@@ -266,6 +304,23 @@ export function validateIndexAccess(
   for (const seg of segments) {
+    // Reject path-injection / illegal characters BEFORE the
+    // membership/pattern/system checks, so this validator is self-contained: its
+    // safety must not depend on a downstream caller URL-encoding the segment. A
+    // `/`, backslash, whitespace, control char, quote, or `< > | %` slipping
+    // through (e.g. in structural-only mode, where membership isn't enforced)
+    // would otherwise enable path injection if interpolated into a request path
+    // unescaped. This is a DENY-list, not an ASCII allow-list: Elasticsearch
+    // permits Unicode index names (e.g. CJK), so a legitimately whitelisted
+    // non-ASCII index must still pass. The wildcard chars `* ?` are deliberately
+    // NOT denied — they're gated by the pattern-membership rule below.
+    if (/[\x00-\x1f/\\\s"'`<>|%]/.test(seg)) {
```

ESLint

[error] Unexpected control character(s) in regular expression: \x00, \x1f.

(no-control-regex)